checks/class-jquery-check.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * Check if JQuery is enqueued on the front end.
+ *
+ * @package Theme Check
+ */
+
+/**
+ * Check if JQuery is enqueued on the front end.
+ */
+class JQuery_Check implements themecheck {
+	/**
+	 * Error messages, warnings and info notices.
+	 *
+	 * @var array $error
+	 */
+	protected $error = array();
+
+	/**
+	 * Check that return true for good/okay/acceptable, false for bad/not-okay/unacceptable.
+	 *
+	 * @param array $php_files File paths and content for PHP files.
+	 * @param array $css_files File paths and content for CSS files.
+	 * @param array $other_files Folder names, file paths and content for other files.
+	 */
+	public function check( $php_files, $css_files, $other_files ) {
+
+		$php = implode( ' ', $php_files );
+
+		checkcount();
+
+		if ( ! preg_match( '/wp_enqueue_script\(\s?("|\')jquery("|\')/i', $php ) ) {
```

Comment:
This will catch enqueueing jQuery successfully. However, it will not catch cases where jQuery is a dependency of another script - in which case it also gets loaded.
Most themes don't enqueue jQuery directly, it is simply used as a dependency for their own scripts, so we should check for that scenario as well.